### PR TITLE
feat: Add support for global deployment file

### LINKED ISSUE #14: [Feature] Global deployment.lua
Hi, thanks for this great plugin!

I was wondering if it is possible to support a global deployment.lua file instead of per project? The location could be in e.g., `vim.fn.stdpath("data")/deployment.lua`. If a local deployment.lua file is present it would overwrite the global configuration. However, it would fallback to the global one.

This would allow users to configure their servers + mappings globally in one place instead of per project.

Is this something you could consider adding as a feature?

### PR DESCRIPTION
**Pull request by smjonas**
Still WIP, feedback appreciated.

Closes #14.

### CODE REVIEW

File: lua/transfer/transfer.lua [hunk 1]
```diff
@@ -183,9 +210,9 @@ function M.remote_scp_path(local_path)
   return nil
 end
 
--- get the remote path for rsync
--- @param local_path string
--- @return string
+---get the remote path for rsync
```
Comment (coffebar) [inline lua/transfer/transfer.lua:213]: What is the reason to change comment style?

File: lua/transfer/transfer.lua [hunk 2]
```diff
@@ -22,9 +41,9 @@ local function reload_buffer(bufnr)
   end
 end
 
--- convert the given local absolute path to a relative project root path
--- @param absolute_path string
--- @return string
+---convert the given local absolute path to a relative project root path
```
Comment (coffebar) [inline lua/transfer/transfer.lua:44]: Let's keep the format "-- comment", not "---comment"

File: lua/transfer/transfer.lua (hunk 1 above)
Comment (smjonas) [reply lua/transfer/transfer.lua:213]: I am using Treesitter parser luap for lua docstrings and they are only highlighted when using `---`:
image

I can revert it if you want though

**Comment by smjonas** [comment]
Tbh, I didn't think that this would require so many changes to the codebase :sweat_smile:. Not sure if I can finish this as I am not too knowledgeable with the existing code base regarding the path handling in `remote_scp_path`..

File: lua/transfer/transfer.lua (hunk 1 above)
Comment (coffebar) [reply lua/transfer/transfer.lua:232]: This is a bug in your highlighting. Treesitter itself and other popular lua plugins use --space in their codebase.

**Comment by smjonas** [comment]
I will close this PR, if someone else wants to implement this feature, feel free to do so using my code

**Pull request closed without merging**